4.13 bug: configure --with-pic=yes passes bad flags to `as`

If you do `./configure --with-pic=yes` then native-code compilation breaks with
```
../boot/ocamlrun ../ocamlopt -strict-sequence -absname -w +a-4-9-41-42-44-45-48-70 -g -warn-error +A -bin-annot -nostdlib -principal -safe-string -strict-formats  -function-sections  -nopervasives -c camlinternalFormatBasics.ml
as: invalid option -- 'P'
File "/home/nojebar/ocaml/stdlib/camlinternalFormatBasics.ml", line 1:
Error: Assembler error, input left in file /tmp/camlasm6a9455.s
Makefile:234: recipe for target 'camlinternalFormatBasics.cmx' failed
make[4]: *** [camlinternalFormatBasics.cmx] Error 2
```

cc @dra27 @shindere

--- a/Changes
+++ b/Changes
@@ -672,7 +672,7 @@ OCaml 4.13.0
 - #10156: configure script: fix sockets feature detection.
   (Lucas Pluvinage, review by David Allsopp and Damien Doligez)
 
-- #10176: By default, call the assembler through the C compiler driver
+- #10176, #10632: By default, call the assembler through the C compiler driver
   (Sébastien Hinderer, review by Gabriel Scherer, David Allsopp and Xavier
   Leroy)
 

--- a/configure
+++ b/configure
@@ -14562,7 +14562,6 @@ if test "$with_pic"; then :
   $as_echo "#define CAML_WITH_FPIC 1" >>confdefs.h
 
   internal_cflags="$internal_cflags $sharedlib_cflags"
-  default_as="$default_as $sharedlib_cflags"
   default_aspp="$default_aspp $sharedlib_cflags"
 else
   fpic=false

--- a/configure.ac
+++ b/configure.ac
@@ -1230,7 +1230,6 @@ AS_IF([test "$with_pic"],
   [fpic=true
   AC_DEFINE([CAML_WITH_FPIC])
   internal_cflags="$internal_cflags $sharedlib_cflags"
-  default_as="$default_as $sharedlib_cflags"
   default_aspp="$default_aspp $sharedlib_cflags"],
   [fpic=false])
 

